### PR TITLE
Install .net 5 for ESRP

### PR DESCRIPTION
**Pull request by dannyvv**
## Description
Manually install .net as needed by ESRP

### Type of Change
Fix pipeline

### Why
.Net 5 is EOL so requires manual force install required by ESPR

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10137)

### CODE REVIEW

File: .ado/jobs/desktop.yml [hunk 1]
```diff
@@ -66,6 +66,17 @@ jobs:
         clean: false
         submodules: false
 
+      # Testing to see if syntax is corerct so we don't have ot wait for
```
Comment (dannyvv) [inline .ado/jobs/desktop.yml:69]: IF this is successfull inthe PR check, I'll remove it as we only need it in the Publish pipeline...

**Comment by dannyvv** [comment]
@jonthysell FYI:
Doing 0.66 only because we need it for office. We'll revisit Monday if we need it for more branches...

**Comment by asklar** [comment]
.net 5 is out of support (.net 6 is the LTS version), can .net 6 be used instead?

**Comment by dannyvv** [comment]
> .net 5 is out of support (.net 6 is the LTS version), can .net 6 be used instead?
Sadly there is no backwards/forwards compatiblity with .net major versions...

The VM's have .Net 6 installed but ESRP is failing: [Logs](https://dev.azure.com/microsoft/ReactNative/_build/results?buildId=51199165&view=logs&j=b6f97fd7-80f3-5748-44f8-8d213965392f&t=3331bb73-cfa5-52db-ad3b-6e163ed96cac) 
```
"C:\Program Files\dotnet\dotnet.exe" <REDACTED>
It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '5.0.0' (x64) was not found.
  - The following frameworks were found:
      2.1.30 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      3.1.18 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      3.1.26 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      6.0.6 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
```

**Review by vmoroz — APPROVED** [review]

**Pull request merged**